core/src/test/java/org/apache/iceberg/avro/TestAvroSchemaProjection.java

```diff
@@ -150,4 +150,58 @@ public void projectWithMapSchemaChanged() {
         .as("Result of buildAvroProjection is missing some IDs")
         .isFalse();
   }
+
+  @Test
+  public void projectWithVariantSchemaChanged() {
+    final org.apache.avro.Schema currentAvroSchema =
+        SchemaBuilder.record("myrecord")
+            .fields()
+            .name("f11")
+            .type()
+            .nullable()
+            .intType()
+            .noDefault()
+            .endRecord();
+
+    final org.apache.avro.Schema variantSchema =
+        SchemaBuilder.record("v")
+            .fields()
+            .name("metadata")
+            .type()
+            .bytesType()
+            .noDefault()
+            .name("value")
+            .type()
+            .bytesType()
+            .noDefault()
+            .endRecord();
+    Variant.get().addToSchema(variantSchema);
+
+    final org.apache.avro.Schema updatedAvroSchema =
+        SchemaBuilder.record("myrecord")
+            .fields()
+            .name("f11")
+            .type()
+            .nullable()
+            .intType()
+            .noDefault()
```

Comment:
If it is nullable, it should have a null default. Maybe that's what happens anyway since that is represented as null/missing?